### DISCUSSION TITLE
Does msquic support multicast?

### DISCUSSION THREAD

**Discussion by cathaysia** [microsoft/msquic#3055] [Q&A] [answered]
Hi, does msquic support multicast?

**Comment by nibanks** [comment, marked as answer]
No it doesn't. QUIC doesn't support it in general. What are you trying to do?

**Reply by cathaysia** [reply]
I want a library that supports reliable multicast, preferably cross-platform. Despite the existence of the PGM protocol, I have not found a good PGM library. Quic, on the other hand, is booming, so I thought I'd ask.

**Reply by nibanks** [reply]
Yeah, lots of time and effort are being put into the QUIC protocol across the industry. There just so happens to be a proposal to add multicast support to QUIC: https://www.ietf.org/id/draft-jholland-quic-multicast-02.html. It's just a draft, and hasn't been officially adopted by the working group; but is a very interesting idea where the multicast part is only from Server to Client, and the client uses its unicast connection to acknowledge received data. We don't have any plans right now to implement it though.

**Reply by Henripru** [reply]
any update on whether this kind of feature may come in the future?

**Reply by nibanks** [reply]
Nothing on my side. If you're interested in this you should follow the IETF, or better yet, reach out to the IETF QUIC WG to ask for/about this.
